src/transforms/remap.rs

```diff
@@ -48,16 +53,37 @@ impl TransformConfig for RemapConfig {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 pub struct Remap {
     program: Program,
     timezone: TimeZone,
     drop_on_error: bool,
     drop_on_abort: bool,
+    enrichment_tables: Arc<ArcSwap<HashMap<String, Box<dyn EnrichmentTable + Send + Sync>>>>,
+}
+
+lazy_static::lazy_static! {
+    static ref MUTEX: std::sync::Mutex<usize> = std::sync::Mutex::new(0);
 }
 
 impl Remap {
-    pub fn new(config: RemapConfig) -> crate::Result<Self> {
+    pub fn new(
+        config: RemapConfig,
+        enrichment_tables: Arc<ArcSwap<HashMap<String, Box<dyn EnrichmentTable + Send + Sync>>>>,
+    ) -> crate::Result<Self> {
+        // Ensure we don't have multiple threads running this code at the same time.
+        let lock = MUTEX.lock().unwrap();
```

Comment:
It's not clear to me from context why we need this to be a critical section.

Reply:
Given that it is swapping out the enrichment tables from the ArcSwap, if there were two remap transforms being compiled simultaneously, one would get the ArcSwap with the tables and the other would get the ArcSwap without the tables and would thus compile incorrectly.

As far as i know we compile the whole config in a single thread, so currently this would never happen. But it's not guaranteed that things will stay that way, so I feel it's best to be safe.

Reply:
Would you mind expanding the comment here to explain that?

Reply:
The code here is not final, ultimately the indexes will be added by VRL during compilation. I'm just largely proving the concept here to make sure we choose the right data structure.

Reply:
I should put a comment in to explain that!